accumulator_updater/programs/accumulator_updater/src/instructions/emit_inputs.rs

```diff
@@ -0,0 +1,165 @@
+use {
+    crate::{
+        accumulator_acc_seeds,
+        accumulator_acc_seeds_with_bump,
+        state::*,
+        AccumulatorUpdaterError,
+    },
+    anchor_lang::{
+        prelude::*,
+        system_program::{
+            self,
+            CreateAccount,
+        },
+    },
+};
+
+
+pub fn emit_inputs<'info>(
+    ctx: Context<'_, '_, '_, 'info, EmitInputs<'info>>,
+    base_account: Pubkey,
+    data: Vec<Vec<u8>>,
+    account_type: u32,
+    account_schemas: Vec<u8>,
+) -> Result<()> {
+    let cpi_caller = ctx.accounts.whitelist_verifier.is_allowed()?;
+    let accts = ctx.remaining_accounts;
+    require_eq!(accts.len(), data.len());
+    require_eq!(data.len(), account_schemas.len());
+    let mut zip = data.into_iter().zip(account_schemas.into_iter());
+
+    let rent = Rent::get()?;
+    let (mut initialized, mut updated) = (vec![], vec![]);
+    for ai in accts {
+        let (account_data, account_schema) = zip.next().unwrap();
+
+        if is_uninitialized_account(ai) {
+            let seeds = accumulator_acc_seeds!(cpi_caller, base_account, account_schema);
+            let (pda, bump) = Pubkey::find_program_address(seeds, &crate::ID);
+            require_keys_eq!(ai.key(), pda);
+
+            //TODO: Update this with serialization logic
+            let accumulator_size = 8 + AccumulatorInput::get_initial_size(&account_data);
+            let accumulator_input = AccumulatorInput::new(
+                AccumulatorHeader::new(
+                    bump,
+                    1, //from CPI caller?
```

Comment:
I think this program controls the header format (and therefore gets to set the version)

I would suggest extracting this value out to a constant like `AccumulatorHeader::CURRENT_VERSION` though so it's easy to bump all the places where it may be used.